src/fides/api/util/tcf/tc_string.py

```diff
@@ -0,0 +1,182 @@
+        bit_components: str = ""
+
+        if isinstance(val, str):
+            # Used for things like publisher country code and consent_language.
+            # There are two letters, represented by 12 bits total, so we convert
+            # the letters to numbers, and they are represented by 6 bits apiece
+            bit_allocation = int(num_bits / len(val))
+            for char in val:
+                converted_num: int = _convert_letter_to_number(char)
+                bit_components += format(converted_num, f"0{bit_allocation}b")
+            return bit_components
+
+        if isinstance(val, list):
+            # List of integers expected.  Bitstring should be the length of the
+            # maximum integer in the list
+            for i in range(1, num_bits + 1):
+                bit_components += "1" if i in val else "0"
+            return bit_components
+
+        # Converts an integer to bits, padding to use the specified number of bits
+        return format(val, f"0{num_bits}b")
+
+    total_bits: str = ""
+    for field in fields:
+        # Either fetch the field of the same name off of the TCModel for encoding,
+        # or use the value override, if supplied.
+        field_value: Any = (
+            field.value_override
+            if field.value_override is not None
+            else getattr(tc_model, field.name)
+        )
+
+        # Cleanup before building bit strings by converting bools to ints
+        if isinstance(field_value, bool):
+            field_value = int(field_value)
+
+        # Converting field value to bitstring of specified length and add that onto the string we're building
+        total_bits += _convert_val_to_bitstring(field_value, field.bits)
+    return total_bits
+
+
+def _convert_letter_to_number(letter: str) -> int:
+    """A->0, Z->25"""
+    return ord(letter) - ord("A")
+
+
+def _convert_bitstring_to_bytes(bitstr: str) -> bytes:
+    """Convert a string of 0's and 1's to bytes, padded to both work with base64
+    and bit->byte conversion"""
+    least_common_multiple = 24  # 6 bits (basis for base 64) and 8 bits (one byte)
+    padding: int = len(bitstr) % least_common_multiple
+    new_bits: str = "0" * (least_common_multiple - padding)
+    bitstr += new_bits
+
+    integer_val: int = int(bitstr, 2)
+    return integer_val.to_bytes((len(bitstr)) // 8, byteorder="big")
+
+
+def _get_max_vendor_id(vendor_list: List[int]) -> int:
+    """Get the maximum vendor id in the supplied list"""
+    if not vendor_list:
+        return 0
+
+    return max(int(vendor_id) for vendor_id in vendor_list)
+
+
+def build_tc_string(model: TCModel) -> str:
+    """Construct a TC String from the given TCModel
+
+    Currently only core and vendors_disclosed sections are supported.
+    """
+    core_string: str = build_core_string(model)
+    vendors_disclosed_string: str = build_disclosed_vendors_string(model)
+
+    return core_string + "." + vendors_disclosed_string
```

Comment:
This is another core new method added here. The ability to encode TC strings on the backend!  This can generically build a TC string from the contents of the TC model. Right now our TC models only describe "accept all" or "reject all" configurations, so in practice, we can just build accept all and reject all strings to start.  But once we can build more targeted tc models, this method should just be able to build the appropriate corresponding string.